src/sortStudents.ts

```diff
@@ -1,16 +1,106 @@
+// import { type } from "os";
 
 export interface Student {
-  // describe Student interface
+  name: string;
+  surname: string;
+  age: number;
+  married: boolean;
+  grades: number[];
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name,
+  Surname,
+  Age,
+  Married,
+  AverageGrade,
 }
 
-// create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
 
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  order: string,
+): Student[] {
+  const copyStudents: Student[] = [...students];
 
-export function sortStudents(students, sortBy, order) {
-  // write your function
+  type Sum = (sum: number, mark: number) => number;
+
+  const sumMarks: Sum = (sum, mark) => sum + mark;
+
+  switch (sortBy) {
+    case SortType.Name:
+      if (order === 'asc') {
+        copyStudents.sort(
+          ({ name: nameA }, { name: nameB }) => nameA.localeCompare(nameB),
+        );
+      } else {
+        copyStudents.sort(
+          ({ name: nameA }, { name: nameB }) => nameB.localeCompare(nameA),
+        );
+      }
+      break;
```

Comment:
```suggestion
  switch (sortBy) {
    case SortType.Name:
      if (order === 'desc') {
       copyStudents.sort(
          ({ name: nameA }, { name: nameB }) => nameB.localeCompare(nameA),
        );
      } 
      
      copyStudents.sort(
        ({ name: nameA }, { name: nameB }) => nameA.localeCompare(nameB),
      );
      
      break;
```
try to simplify the code like this

Reply:
variable works here too